src/Microsoft.SqlTools.ServiceLayer/DacFx/ExtractOperation.cs

```diff
@@ -37,7 +37,7 @@ public override void Execute()
             }
             catch
             {
-                Logger.Write(TraceEventType.Warning, string.Format(CultureInfo.InvariantCulture, "Invalid version {0} passed", this.Parameters.ApplicationVersion));
+                throw new Exception(string.Format("Invalid version {0} passed. Version must be in the format x.x.x.x where x is a number", this.Parameters.ApplicationVersion));
```

Comment:
I throwing a generic exception right here? Consider an ArgumentException of an InvalidOperationException.

Reply:
changed it to be an ArgumentException